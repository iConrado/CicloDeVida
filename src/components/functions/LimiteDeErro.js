import React from 'react';
import { Text } from 'react-native';

import styles from './styles';

export default class LimiteDeErro extends React.Component {
  constructor(props) {
    super(props);
    this.state = { hasError: false };
  }

  componentDidCatch(error, info) {
    // Display fallback UI
    this.setState({ hasError: true });
    // You can also log the error to an error reporting service
    // logErrorToMyService(error, info);
    // console.log(error, info);
  }

  render() {
    if (this.state.hasError) {
      // You can render any custom fallback UI
      return <Text style={styles.label}>Um erro aconteceu...</Text>;
    }
    return this.props.children; //eslint-disable-line
  }
}
