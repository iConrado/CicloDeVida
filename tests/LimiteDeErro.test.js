import React from 'react';
import { shallow } from 'enzyme';
import { Text } from 'react-native';
import LimiteDeErro from '../src/components/functions/LimiteDeErro';


const app = shallow(<LimiteDeErro />);

describe('LimiteDeErro', () => {
  it('renderiza corretamente', () => {
    expect(app).toMatchSnapshot();
  });

  it('renderiza com erro', () => {
    const func = jest.fn(() => {
      app.instance().componentDidCatch({ error: 'erro' }, { info: 'info erro' });
    });
    func();
    expect(app.state('hasError')).toBe(true);
  });
});
