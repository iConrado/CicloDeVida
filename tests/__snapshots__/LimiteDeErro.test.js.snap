// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LimiteDeErro renderiza corretamente 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LimiteDeErro />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): undefined,
  Symbol(enzyme.__nodes__): Array [
    undefined,
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
